hack/generated/Makefile

```diff
@@ -70,16 +65,52 @@ generate: $(CONTROLLER_GEN) $(CONVERSION_GEN) ## Generate code
 #		--go-header-file=../boilerplate.go.txt
 
 	# Force regeneration of all of the CRDs
-	@echo "Deleting old CRDs"
+	@echo "Deleting old CRD YAMLs"
 	$(Q) if [ -d "./config/crd/bases" ]; then find "./config/crd/bases" -type f -name "*" -delete; fi
 
-	@echo "Executing controller-gen to generate CRDs"
-	$(Q) $(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./apis/..." output:crd:artifacts:config=config/crd/bases
+	@echo "Executing controller-gen to generate CRD and RBAC YAMLs"
+	$(Q) $(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
 
 .PHONY: build
 build: tidy lint ; $(info $(M) building ./bin/$(APP))
 	$(Q) $(GO) build -o ./bin/$(APP)
 
+## --------------------------------------
+## Test
+## --------------------------------------
+.PHONY: test
+test: ; $(info $(M) running go test…)
+	$(Q) $(GO) test $(GO_DIRS_TO_TEST) -short -tags=noexit
+
+# Initially this target uses kind as there is no need to end to end test on Azure.
+# Eventually when we start supporting features like MSI that only work in Azure,
+# it's likely we'll need to make the integration tests (or at least some of them)
+# run in an AKS cluster.
+.PHONY: test-int-no-cleanup
+test-int-no-cleanup: $(ROOT_DIR)/.env ; $(info $(M) running controller integration test…)
+	# MUST be executed as single command, or env vars will not propagate to test execution
+	$(Q) . $(ROOT_DIR)/.env && $(GO) test $(GO_DIRS_TO_TEST) # TODO: better way to just run integration tests?
+
+.PHONY: test-int
+test-int: kind-create deploy test-int-no-cleanup kind-delete cleanup-test-azure-resources
```

Comment:
we really are not using Make as intended but as a series of commands. perhaps we should look into an alternate solution

Reply:
Agreed -- file an item for it maybe?